includes/classes/InternalConnections/NetworkSiteConnection.php

```diff
@@ -751,6 +674,102 @@ public static function get_available_authorized_sites( $context = null ) {
 		return apply_filters( 'dt_authorized_sites', $authorized_sites, $context );
 	}
 
+	/**
+	 * Build the available sites a specific user is authorized to use.
+	 *
+	 * @param int|string $user_id Current user ID
+	 * @param string     $context The context of the authorization. Either push or pull
+	 * @param bool       $force Force a cache clear. Default false
+	 *
+	 * @return array
+	 */
+	public static function build_available_authorized_sites( $user_id = '', $context = null, $force = false ) {
+		$user_id      = ! $user_id ? get_current_user_id() : $user_id;
+		$last_changed = get_site_option( 'last_changed_sites' );
+
+		if ( ! $last_changed ) {
+			$last_changed = microtime();
+			self::set_sites_last_changed_time();
+		}
+
+		$cache_key        = "authorized_sites:$user_id:$context:$last_changed";
+		$authorized_sites = get_transient( $cache_key );
+
+		if ( $force || false === $authorized_sites ) {
+			$sites           = get_sites();
```

Comment:
Any reason we only get 100 sites? seems like we could handle far more. Maybe even make this filterable? 
```suggestion
			$sites           = $sites           = get_sites( array (
				'number' => 1000,
			));
```

Reply:
Bumped this up to 1000